### PR TITLE
Adding develepor options to add fake media

### LINKED ISSUE #13480: Add a way to simulate/test storage migration on device
**Is your feature request related to a problem? Please describe.**

To test storage migration, I have to:
* Go to Settings > Advanced > AnkiDroid directory, and set it to a public folder
* Copy some files into `collection.media` of that folder

This is annoying and takes time and requires a computer (or a lot of tapping)

**Describe the solution you'd like**

A menu option to create a dummy collection + media in a public folder, perhaps with options to specify file number and sizes

### PR DESCRIPTION
**Pull request by Arthur-Milchior**
This will add to manually test migration of media.

2023-05-16-010442_401x701_scrot

Bug: #13480 from @oakkitten

### CODE REVIEW

**Review by david-allison — DISMISSED** [review]
Lint is failing

Primary issue:

Needs to be in "Developer Options" rather than "Advanced"

Aside: There's little signal if we're testing the same count/size of files every time. Might as well be an apkg

**Comment by oakkitten** [comment]
I think this needs to be a more easily accessible action, e.g. a *main menu* option to “Create a fake deck” or something with the options to set:

* Collection location
* Number and sizes of media files to generate (e.g. 10-20 files of 1-10MB each)
* Whether or not some media files are referenced to by cards or not
* Maybe something else?

Also I'd love to have the option to crash during migration

Also I'd love to have the option to reset migration state

File: AnkiDroid/src/main/java/com/ichi2/anki/dialogs/DeckPickerContextMenu.kt [hunk 1]
```diff
@@ -123,7 +123,7 @@ class DeckPickerContextMenu(private val collection: Collection) : AnalyticsDialo
             DeckPickerContextMenuOption.RENAME_DECK -> {
                 Timber.i("Rename deck selected")
                 (activity as DeckPicker).renameDeckDialog(deckId)
-                (activity as AnkiActivity).dismissAllDialogFragments()
```
[the remark this answers is not in the document]
Comment (Arthur-Milchior) [reply AnkiDroid/src/main/java/com/ichi2/anki/dialogs/DeckPickerContextMenu.kt:126]: Visibly something remaining from a different PR. Strange

**Comment by Arthur-Milchior** [comment]
If that's okay with you, @oakkitten, this PR will deal with media creation. 

I am not sure what you mean by "main menu".
The trouble I had with putting this action in DeckPicker is that I really don't want that a user find it by accident. With developer or advanced, there is no risk here.
I could ensure it appears only if you are in DEBUG mode, but that's not great to test the version on the play store then

**Comment by oakkitten** [comment]
The overflow menu in deck picker could have a menu item “Debug” opening a submenu with some options. It could be enabled either via `DEBUG` or via a setting in Developer options, e.g. “Show debug menu”

> If that's okay with you, @oakkitten, this PR will deal with media creation.

I mean, I don't always fill the media folder with random files when testing. The `MigrateUserData` task will migrate *some* files so you will see if it is broken at all. I would like to be able to quickly set the collection to a public storage though, that's more useful. And I would like to be able to reset the migration—if it fails, and keeps failing, you can't quite escape the loop (by the way, we do not have a way of dealing with this!) and the only way to resolve the situation is to ”Delete everything” in “Manage space activity”, which has an additional effect of voiding all of the permissions including the ones you must set via adb/some advanced preferences.

P.S. I imagine that the settings in the “Create a fake deck” dialog would be saved into shared preferences.

**Comment by Arthur-Milchior** [comment]
So, right now, I put the action in the developer settings. If needed it can be moved later. 
In particular, I don't expect to need to use it that much that putting it in main instead of in settings will change that much.
I suggest we add more option to help manual test later, in other PR, so that at least this can be used.

File: AnkiDroid/src/main/java/com/ichi2/anki/preferences/DevOptionsFragment.kt [hunk 2]
```diff
@@ -78,6 +80,46 @@ class DevOptionsFragment : SettingsFragment() {
             OnboardingUtils.reset(requireContext())
             true
         }
+
+        val sizePreference = requirePreference<IncrementerNumberRangePreferenceCompat>(getString(R.string.pref_fill_collection_size_file))
+        val numberOfFilePreference = requirePreference<IncrementerNumberRangePreferenceCompat>(getString(R.string.pref_fill_collection_number_file))
+
+        /*
+         * Debugging section
```
Comment (BrayanDSO) [inline AnkiDroid/src/main/java/com/ichi2/anki/preferences/DevOptionsFragment.kt:88]: the section isn't called "Debugging".

File: AnkiDroid/src/main/java/com/ichi2/anki/preferences/DevOptionsFragment.kt [hunk 3]
```diff
@@ -78,6 +80,46 @@ class DevOptionsFragment : SettingsFragment() {
...
+         */
+        requirePreference<Preference>(R.string.pref_fill_collection).setOnPreferenceClickListener {
+            val sizeOfFiles = sizePreference.getValue()
+            val numberOfFiles = numberOfFilePreference.getValue()
+            AlertDialog.Builder(requireContext()).show {
+                setTitle("Warning!")
+                setMessage("You'll add $numberOfFilePreference files with no meaningful content, potentially overriding existing files. Do not do it on a collection you care about.")
```
Comment (BrayanDSO) [inline AnkiDroid/src/main/java/com/ichi2/anki/preferences/DevOptionsFragment.kt:95]: $numberOfFilePreference will print the title of the preference

File: AnkiDroid/src/main/java/com/ichi2/anki/preferences/DevOptionsFragment.kt [hunk 4]
```diff
@@ -78,6 +80,46 @@ class DevOptionsFragment : SettingsFragment() {
...
+                val suffix = ".$size"
+                for (i in 1..numberOfFiles) {
+                    val f = File.createTempFile("00$i", suffix)
+                    f.appendBytes(ByteArray(size))
+
+                    CollectionManager.withCol {
+                        media.addFile(f)
+                    }
+                    if (i % 1000 == 0) {
+                        showSnackbar("$i files added.")
```
Comment (BrayanDSO) [inline AnkiDroid/src/main/java/com/ichi2/anki/preferences/DevOptionsFragment.kt:118]: can't see that snackbar showing up

File: AnkiDroid/src/main/java/com/ichi2/anki/preferences/DevOptionsFragment.kt [hunk 5]
```diff
@@ -78,6 +80,46 @@ class DevOptionsFragment : SettingsFragment() {
...
+        }
+    }
+
+    private fun generateFiles(size: Int, numberOfFiles: Int) {
+        launchCatchingTask {
+            withProgress("Generating $numberOfFiles files of size $size bytes") {
+                // Add 10.000 small files
+                val suffix = ".$size"
+                for (i in 1..numberOfFiles) {
+                    val f = File.createTempFile("00$i", suffix)
```
Comment (BrayanDSO) [inline AnkiDroid/src/main/java/com/ichi2/anki/preferences/DevOptionsFragment.kt:111]: this gives me an IDE warning, either suppress (with a justification comment) it or handle with it

File: AnkiDroid/src/main/res/values/preferences.xml [hunk 6]
```diff
@@ -113,6 +113,9 @@
     <string name="card_browser_external_context_menu_key">card_browser_enable_external_context_menu</string>
     <string name="anki_card_external_context_menu_key">anki_card_enable_external_context_menu</string>
     <string name="pref_reset_languages_key">resetLanguages</string>
+    <string name="pref_fill_collection">fillCollection</string>
+    <string name="pref_fill_collection_number_file">fillCollectionNumberFile</string>
+    <string name="pref_fill_collection_size_file">fillCollectionSizeFile</string>
```
Comment (BrayanDSO) [inline AnkiDroid/src/main/res/values/preferences.xml:118]: nit: put these together with the other dev options keys, i.e. below `<!-- Developer options -->` in this same file

File: AnkiDroid/src/main/res/xml/preferences_dev_options.xml [hunk 7]
```diff
@@ -51,4 +53,28 @@
         android:title="@string/reset_onboarding"
         android:summary="@string/reset_onboarding_desc"
         android:key="@string/pref_reset_onboarding_key"/>
+    <PreferenceCategory
+        android:title="Create medias"
```
Comment (BrayanDSO) [inline AnkiDroid/src/main/res/xml/preferences_dev_options.xml:57]: I believe that media is a plural word.

**Review by BrayanDSO — CHANGES_REQUESTED** [review]

File: AnkiDroid/src/main/res/values/preferences.xml (hunk 6 above)
Comment (david-allison) [reply AnkiDroid/src/main/res/values/preferences.xml:118]: We don't need any strings. Keep the debug settings in English

Comment (BrayanDSO) [reply AnkiDroid/src/main/res/values/preferences.xml:118]: these are the keys constants, not strings

Comment (david-allison) [reply AnkiDroid/src/main/res/values/preferences.xml:118]: That's what I get for reviewing on my phone, thanks!

**Review by BrayanDSO — APPROVED** [review]
Took the liberty to fix myself the issues I found to avoid getting stale, since this is pretty useful.

**Review by lukstbit — APPROVED** [review]
LGTM
Note that I added some minor changes to improve the names used and to fix an issue with the last toast shown.

**Pull request merged**